### PR TITLE
feat: honor DO_NOT_TRACK as a metrics opt-out

### PR DESCRIPTION
**Pull request by heymatthew**
## What

Honor the cross-tool [`DO_NOT_TRACK`](https://donottrack.sh/) standard as an alias for `BD_DISABLE_METRICS`, so a single environment variable used across many dev tools opts the user out of bd's anonymous usage metrics.

## How

- `internal/metrics`: add the `EnvDoNotTrack = "DO_NOT_TRACK"` constant.
- `resolveMetricsEnabled()`: consult `DO_NOT_TRACK` with the same truthy semantics as `BD_DISABLE_METRICS`, **only when `BD_DISABLE_METRICS` is unset** — the bd-specific variable always takes precedence when both are set. `DO_NOT_TRACK` is an env signal honored at runtime; it does not write config.
- `bd metrics` status and the on/off warnings surface whichever override is active (extracted a shared `metricsEnvOverride()` helper preserving that precedence).
- Docs: `AGENT_INSTRUCTIONS.md` documents the alias and cites the standard.

## Truth Table

| `BD_DISABLE_METRICS` | `DO_NOT_TRACK` | Metrics |
| -------------------- | -------------- | ------- |
| unset                | unset          | config default (on) |
| unset                | `1` / truthy   | **off** |
| unset                | `0` / empty    | on      |
| `0`                  | `1`            | **on** (bd var wins) |
| `1`                  | `0`            | **off** (bd var wins) |

## Tests

- `TestResolveMetricsEnabledHonorsDoNotTrack`: table-driven precedence + truthy-value coverage.
- Made the end-to-end metrics tests hermetic: they now strip `DO_NOT_TRACK` from the spawned-`bd` env alongside `BD_DISABLE_METRICS`, so an ambient opt-out on a developer/CI machine can no longer disable the metrics they assert are emitted. (This was a latent gap the new behavior would otherwise expose.)

Verified with `go test -tags gms_pure_go ./cmd/bd/ ./internal/metrics/...` (including a run with `DO_NOT_TRACK=1` set in the ambient env to prove hermeticity) and `go vet`.

### CODE REVIEW

File: cmd/bd/init_metrics_test.go [hunk 1]
```diff
@@ -86,7 +86,10 @@ func metricsTestEnv(home string, extra ...string) []string {
 	base := bdEnv(home)
 	out := make([]string, 0, len(base)+len(extra)+1)
 	for _, e := range base {
-		if strings.HasPrefix(e, "BD_DISABLE_METRICS=") {
+		// Strip both metrics opt-out env vars so an ambient DO_NOT_TRACK or
+		// BD_DISABLE_METRICS on the developer/CI machine can't disable the
+		// metrics these end-to-end tests assert are emitted.
```
Comment (heymatthew) [inline cmd/bd/init_metrics_test.go]: Minimize inline code comments. ([CONTRIBUTING_PR_GUIDELINES.md](https://github.com/gastownhall/beads/blob/main/CONTRIBUTING_PR_GUIDELINES.md))

**Comment by julianknutsen** [comment]
## Maintainer Adoption Review

Thanks for the contribution, @heymatthew! This PR honors the cross-tool [`DO_NOT_TRACK`](https://donottrack.sh/) standard as a runtime alias for `BD_DISABLE_METRICS`, so the single environment variable many developers already export opts them out of bd's anonymous usage metrics — a privacy-positive default that meets people where their tooling already is.

This PR was reviewed and adopted with maintainer fixes pushed directly to the PR branch.

### Original PR Review
Decision: approve.

Specific gaps fixed:
- The first alias implementation treated `DO_NOT_TRACK` as a bidirectional override, so a falsey-but-present value (e.g. `DO_NOT_TRACK=0`) could re-enable metrics even after a user had explicitly saved `bd metrics off`. The final patch makes `DO_NOT_TRACK` **disable-only**: a truthy value opts out, while falsey/empty values fall through to saved config, and `BD_DISABLE_METRICS` stays the higher-precedence override when both are set.
- The override-messaging refactor (status output and the on/off warnings naming the active env var) was previously unverified. The final patch adds focused coverage so `DO_NOT_TRACK`-alone and both-vars-set cases assert the reported override and preserved `BD_DISABLE_METRICS` precedence.

Review findings addressed:
- Contract & interface fidelity (major, medium confidence, Claude): the alias could force metrics on over a saved opt-out; the final patch restores disable-only semantics so the privacy signal can never re-enable telemetry.
- Test evidence quality (minor, high confidence, Claude): the review found the new override messaging untested; the final patch adds table-driven tests covering truthy/falsey/empty `DO_NOT_TRACK`, saved-off config, and `BD_DISABLE_METRICS` precedence, and makes the end-to-end metrics tests hermetic against an ambient `DO_NOT_TRACK`.

Remaining non-gating notes:
- No remaining gating issues were found. The Gemini reviewer was skipped by this run's configuration; Claude and Codex reviewed independently and agreed on the disable-only behavior, override reporting, and regression coverage.

### Maintainer Changes
`fix(metrics): make DO_NOT_TRACK a disable-only opt-out` — 4 files changed, 189 insertions(+), 11 deletions(-) (`AGENT_INSTRUCTIONS.md`, `cmd/bd/main.go`, `cmd/bd/metrics.go`, `cmd/bd/metrics_test.go`). Contributor authorship on the original commit is preserved.

### Final Review Status
Ready for the merge queue: final head `7b5ffc5b2` has passing required GitHub checks, and the merge-ready metadata is in place. I am marking this PR `status/merge-ready` so the merge queue can pick it up.

CI: https://github.com/gastownhall/beads/actions/runs/29839472776

### Review Iterations
2 review passes performed: the first requested changes for the disable-only semantics and test coverage; after the maintainer fix, the second pass approved with no new findings.

---
_Adopted via `/adopt-pr` workflow. Original contributor commits preserved._
